### ISSUE TITLE
Azure service operator for cosmos db collection and index creation or update

### ISSUE THREAD

**Issue by vivuu1989** [Azure/azure-service-operator#3208]
We were trying to create azure operator for managing our existing azure cosmos db dor managing and administration the collection or index creation or update as a self service for our developers, so that they can create these resourcs on demand using azuredevops. But we couldn't find any solution for this with azure operator.

**Comment by theunrepentantgeek** [comment]
Reviewing the Azure documentation for [SQL Databases](https://learn.microsoft.com/en-us/azure/templates/microsoft.documentdb/databaseaccounts/sqldatabases?pivots=deployment-language-arm-template) and [Tables](https://learn.microsoft.com/en-us/azure/templates/microsoft.documentdb/databaseaccounts/tables?pivots=deployment-language-arm-template), it doesn't appear that collections and indexes are ARM resources, so we can't use our code generator to do most of the work required for support. 

Instead, they seem to require use of the relevant Cosmos SDK.

I'll tag this issue appropriately and we'll do some work to estimate the effort required so that we can prioritize this.

**Comment by theunrepentantgeek** [comment]
We discussed this issue in today's weekly ASO sync meeting. We're interested in supporting collections and indexes as child resources, but we're not sure where it falls in our current backlog. For the moment, we've added it to our v2.4.0 milestone.

That said, we do accept PRs and we're always happy to work with people who want to contribute to ASO. If you'd like to author these resources yourself, you're most welcome. A good place to start would be to review [PostgreSQL users](https://github.com/Azure/azure-service-operator/blob/main/v2/api/dbforpostgresql/v1/user_types.go), as we hand wrote those ourselves.
